packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { BaseComponent, classNamesFunction, customizable, DelayedRender } from '../../Utilities';
+import { IShimmerProps, IShimmerStyleProps, IShimmerStyles } from './Shimmer.types';
+import { ShimmerElementsGroup } from './ShimmerElementsGroup/ShimmerElementsGroup';
+
+export interface IShimmerState {
+  /**
+   * Flag for knowing when to remove the shimmerWrapper from the DOM.
+   */
+  contentLoaded?: boolean;
+}
+
+const TRANSITION_ANIMATION_INTERVAL = 200; /* ms */
+
+const getClassNames = classNamesFunction<IShimmerStyleProps, IShimmerStyles>();
+
+@customizable('Shimmer', ['theme'])
+export class ShimmerBase extends BaseComponent<IShimmerProps, IShimmerState> {
+  public static defaultProps: IShimmerProps = {
+    isDataLoaded: false
+  };
+
+  private _classNames: { [key in keyof IShimmerStyles]: string };
+  private _lastTimeoutId: number | undefined;
+
+  constructor(props: IShimmerProps) {
+    super(props);
+
+    this.state = {
+      contentLoaded: props.isDataLoaded
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IShimmerProps): void {
+    const { isDataLoaded } = nextProps;
+
+    if (this._lastTimeoutId !== undefined) {
+      this._async.clearTimeout(this._lastTimeoutId);
+      this._lastTimeoutId = undefined;
+    }
+    if (isDataLoaded) {
+      this._lastTimeoutId = this._async.setTimeout(() => {
+        this.setState({
+          contentLoaded: isDataLoaded
+        });
+        this._lastTimeoutId = undefined;
+      }, TRANSITION_ANIMATION_INTERVAL);
+    } else {
+      this.setState({
+        contentLoaded: isDataLoaded
+      });
+    }
+  }
+
+  public render(): JSX.Element {
+    const {
+      styles,
+      shimmerElements,
+      children,
+      isDataLoaded,
+      width,
+      className,
+      customElementsGroup,
+      theme,
+      ariaLabel
+    } = this.props;
+
+    const { contentLoaded } = this.state;
+
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      isDataLoaded,
+      className,
+      transitionAnimationInterval: TRANSITION_ANIMATION_INTERVAL
+    });
+
+    return (
+      <div className={this._classNames.root}>
```

Comment:
You should mix in the ...rest. Or native props. If you stick something like an inline style on this, or a data attribute, or an id, it won't work, even though you accept those things. Can you add that?

Reply:
@dzearing I hope I understood what you meant by "mix in the ...rest" and I added getNativeProps of divProperties.